browser/src/controller.js

```diff
@@ -0,0 +1,174 @@
+import { interpret } from 'xstate';
+import $ from 'jquery';
+
+import { deepcellLabelMachine } from './statechart';
+import { Model } from './model';
+import { History } from './history';
+import keydownLookup from './keydownLookup';
+
+export class Controller {
+  /**
+   * Retrives a Project, and sets up bindings to control Label.
+   * @param {string} projectID 12 character base64 ID for Project in DeepCell Label database
+   */
+  constructor(projectID) {
+    // Get Project from database
+    const getProject = $.ajax({
+      type: 'GET',
+      url: `${document.location.origin}/api/project/${projectID}`,
+      async: true
+    });
+
+    // Create model and view for Project and setup bindings
+    getProject.done((project) => {
+      this.model = new Model(project);
+      this.view = this.model.view;
+      this.history = new History();
+
+      // Interpret the statechart
+      this.service = interpret(deepcellLabelMachine);
+      // add a listener to update the info table whenever a transition occurs
+      this.service.onTransition(() => { this.model.notifyInfoChange() });
+      // Start the service
+      this.service.start();
+
+      // Enable global access
+      window.model = this.model;
+      window.view = this.view;
+      window.service = this.service;
+
+      // Add bindings
+      this.overrideScroll();
+      this.addWindowBindings();
+      this.addCanvasBindings();
+      this.addUndoBindings();
+    });
+  }
+
+  undo() {
+    this.history.undo();
+    this.model.updateMousePos(this.model.canvas.rawX, this.model.canvas.rawY);
+  }
+
+  redo() {
+    this.history.redo();
+    this.model.updateMousePos(this.model.canvas.rawX, this.model.canvas.rawY);
+  }
+
+  /**
+   * Disable scrolling with the wheel (on the canvas) or with keys (everywhere)
+   */
+  overrideScroll() {
+    const canvasElement = document.getElementById('canvas');
+    canvasElement.onwheel = (event) => { event.preventDefault(); };
+    canvasElement.onmousewheel = (event) => { event.preventDefault(); };
```

Comment:
With single line arrow functions, you can simplify this!

```suggestion
    canvasElement.onwheel = event => event.preventDefault();
    canvasElement.onmousewheel = event => event.preventDefault();
```

Reply:
Fixed here and for other single parameter arrow functions in this module. I noticed that we have mixed parameter names for event handlers `event =>`, `evt =>`, and `e =>`. Should we choose one and stick with it throughout?

Reply:
Yeah, I think it would be nice to have consistent variable naming throughout the project.